### PR TITLE
feat(earn): add function for fetching aave rewards

### PR DESCRIPTION
**Pull request by satish-ravi**
### Description

As the title

### Test plan

Unit tests

### Related issues

- Part of ACT-1180

### Backwards compatibility

Yes

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [x] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

### CODE REVIEW

File: src/web3/networkConfig.ts [hunk 1]
```diff
@@ -154,6 +155,10 @@ const AAVE_ARB_USDC_TOKEN_ID_MAINNET = `${NetworkId['arbitrum-one']}:0x724dc807b
 const ARB_AAVE_POOL_V3_CONTRACT_ADDRESS_STAGING = '0xBfC91D59fdAA134A4ED45f7B584cAf96D7792Eff'
 const ARB_AAVE_POOL_V3_CONTRACT_ADDRESS_MAINNET = '0x794a61358D6845594F94dc1DB02A252b5b4814aD'
 
+// also called rewards controller
+const ARB_AAVE_INCENTIVES_V3_CONTRACT_ADDRESS_STAGING = '0x3A203B14CF8749a1e3b7314c6c49004B77Ee667A'
```
Comment (MuckT) [inline src/web3/networkConfig.ts:159]: These should be lowercase for consistency.

File: src/abis/AaveIncentivesV3.ts
Comment (MuckT) [inline src/abis/AaveIncentivesV3.ts:1]: Would be nice to have a comment as to where we can find this abi.

File: src/earn/poolInfo.ts [hunk 2]
```diff
@@ -43,3 +47,51 @@ export async function fetchAavePoolInfo({
...
+    Logger.debug(TAG, 'Fetching Aave reward info', {
+      assetAddress,
+      contractAddress,
+      network,
+    })
+    const result = await publicClient[network].readContract({
+      abi: aaveIncentivesV3Abi,
+      address: contractAddress,
+      functionName: 'getAllUserRewards',
+      args: [[assetAddress], walletAddress],
```
Comment (MuckT) [inline src/earn/poolInfo.ts:75]: Is this nested array expected?

File: src/web3/networkConfig.ts (hunk 1 above)
Comment (satish-ravi) [reply src/web3/networkConfig.ts:159]: I believe contract addresses in the file are currently all checksummed addresses

File: src/earn/poolInfo.ts (hunk 2 above)
Comment (satish-ravi) [reply src/earn/poolInfo.ts:75]: yes, the first argument to the contract method is an array

File: src/abis/AaveIncentivesV3.ts
Comment (satish-ravi) [reply src/abis/AaveIncentivesV3.ts:1]: added a link to aave's docs

**Review by MuckT — APPROVED** [review]

File: src/earn/poolInfo.ts [hunk 3]
```diff
@@ -43,3 +47,51 @@ export async function fetchAavePoolInfo({
...
+  allTokens: TokenBalance[]
+}) {
+  try {
+    const network = networkIdToNetwork[networkId]
+    Logger.debug(TAG, 'Fetching Aave reward info', {
+      assetAddress,
+      contractAddress,
+      network,
+    })
+    const result = await publicClient[network].readContract({
```
Comment (silasbw) [inline src/earn/poolInfo.ts:71]: nit: can you destructure this result instead of doing result[0] and result[1]?

**Review by silasbw — APPROVED** [review]

**Pull request merged**